src/toolkits/nearest_neighbors/nearest_neighbors.cpp

```diff
@@ -318,6 +318,25 @@ void nearest_neighbors_model::train(const sframe& X,
   train(X, ref_labels, composite_distance_params, opts);
 }
 
+void nearest_neighbors_model::populate_distance_for_summary_struct(
+    const std::vector<dist_component_type>& composite_distance_params) {
+  if (composite_distance_params.size() > 1) {
+    add_or_update_state({
+      {"distance_for_summary_struct", "composite"}
+    });
+  } else if (composite_distance_params.size() == 1 
+      && std::get<0>(composite_distance_params[0]).size() > 1) {
+    add_or_update_state({
+      {"distance_for_summary_struct", "composite"}
+    });
+  } else {
+    function_closure_info distance_fn = std::get<1>(composite_params[0]);
+    auto dist_name = extract_distance_function_name(distance_fn);
```

Comment:
I think should be `std::string` rather than `auto`.

Reply:
Done.